pkg/cmd/build/build_test.go

```diff
@@ -236,3 +240,127 @@ func Test_OptsFromBuildInfo(t *testing.T) {
 		})
 	}
 }
+
+func TestOptsFromBuildInfo(t *testing.T) {
+	buildName := "frontendTest"
+	mockDir := "mockDir"
+
+	originalWd, errwd := os.Getwd()
+	if errwd != nil {
+		t.Fatal(errwd)
+	}
+
+	dir := t.TempDir()
+	log.Printf("created tempdir: %s", dir)
+
+	os.Chdir(dir)
+	defer os.Chdir(originalWd)
+
+	err := os.Mkdir(buildName, os.ModePerm)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = os.Mkdir(mockDir, os.ModePerm)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	contextPath := filepath.Join(dir, buildName)
+	log.Printf("created context dir: %s", contextPath)
+
+	tests := []struct {
+		name               string
+		svcName            string
+		dockerfile         string
+		fileExpected       string
+		dockerfilesCreated []string
+		expectedError      string
+	}{
+		{
+			name:               "dockerfile is abs path",
+			svcName:            "t1",
+			dockerfile:         filepath.Join(contextPath, "Dockerfile"),
+			fileExpected:       filepath.Join(contextPath, "Dockerfile"),
+			dockerfilesCreated: nil,
+			expectedError:      "",
+		},
+		{
+			name:               "dockerfile is NOT relative to context",
+			svcName:            "t2",
+			dockerfile:         "Dockerfile",
+			fileExpected:       "Dockerfile",
+			dockerfilesCreated: []string{"Dockerfile"},
+			expectedError:      fmt.Sprintf(warningDockerfilePath, "t2", "Dockerfile", buildName),
+		},
+		{
+			name:               "dockerfile in root and dockerfile in context path",
+			svcName:            "t3",
+			dockerfile:         "Dockerfile",
+			fileExpected:       filepath.Join(buildName, "Dockerfile"),
+			dockerfilesCreated: []string{"Dockerfile", filepath.Join(buildName, "Dockerfile")},
+			expectedError:      fmt.Sprintf(doubleDockerfileWarning, "t3", buildName, "Dockerfile"),
+		},
+		{
+			name:               "dockerfile is relative to context",
+			svcName:            "t4",
+			dockerfile:         "Dockerfile",
+			fileExpected:       filepath.Join(buildName, "Dockerfile"),
+			dockerfilesCreated: []string{filepath.Join(buildName, "Dockerfile")},
+			expectedError:      "",
+		},
+	}
+
+	for _, tt := range tests {
+
+		t.Run(tt.name, func(t *testing.T) {
+
+			if tt.dockerfilesCreated != nil {
+				for _, df := range tt.dockerfilesCreated {
+					defer removeFile(df)
+					_, err := os.Create(df)
+					if err != nil {
+						t.Fatal(err)
+					}
+
+					log.Printf("created docker file: %s", df)
+				}
+			}
+
+			var buf bytes.Buffer
+			oktetoLog.SetOutput(&buf)
+
+			defer func() {
+				oktetoLog.SetOutput(os.Stderr)
+			}()
+
+			file := extractFromContextAndDockerfile(buildName, tt.dockerfile, tt.svcName)
+			warningErr := strings.TrimSuffix(buf.String(), "\n")
+
+			if warningErr != "" && tt.expectedError == "" {
```

Comment:
Not needed to address in the PR but as a general comment. In general, I would try to avoid to include ifs depending on input parameters for tests. This makes tests more difficult to read, understand and more error prone. 

`table driven test` is really good when you want to test different scenarios  of a function with different inputs and outputs but without having to include branching code in the tests.